### PR TITLE
Remove deprecation warning in PostController

### PR DESCRIPTION
**Pull request by core23**

### CODE REVIEW

**Comment by sagikazarmark** [comment]
:+1:

**Comment by OskarStark** [comment]
:+1:

File: Controller/PostController.php [hunk 1]
```diff
@@ -30,16 +31,17 @@ public function homeAction()
...
-     * @param array $criteria
-     * @param array $parameters
+     * @param array   $criteria
+     * @param array   $parameters
+     * @param Request $request
      *
      * @return \Symfony\Component\HttpFoundation\Response
      */
-    public function renderArchive(array $criteria = array(), array $parameters = array())
+    public function renderArchive(array $criteria = array(), array $parameters = array(), Request $request)
```
Comment (Soullivaneuh) [inline Controller/PostController.php]: This BC break.

Please see `CRUDController` from admin-bundle to see how this is handled.

Comment (Soullivaneuh) [inline Controller/PostController.php]: You can maybe call `resolveReques` as done here: https://github.com/sonata-project/SonataNewsBundle/pull/257/files#diff-470b4c7629c3b1960c1c3978c9069fd9R72

**Comment by core23** [comment]
Added BC info to changelog file and refactored the ``renderAction`` method

**Comment by Soullivaneuh** [comment]
> Added BC info to changelog file and refactored the renderAction method

There is no BC break info to put if this is handled.

**Comment by core23** [comment]
Sorry, I was too enthusiastic adding changelog notes ;)

**Comment by core23** [comment]
@Soullivaneuh ready to merge :+1:

File: Controller/PostController.php [hunk 2]
```diff
@@ -107,38 +121,45 @@ public function collectionAction($collection)
...
             'date' => $this->getPostManager()->getPublicationDateQueryParts(sprintf('%d-%d-%d', $year, 1, 1), 'year'),
-        ));
+        ), $request);
     }
 
     /**
      * @throws NotFoundHttpException
      *
      * @param $permalink
+     * @param Request $request
```
Comment (OskarStark) [inline Controller/PostController.php]: you added a phpdoc, but you don't add ``$request`` to the signature of ``viewAction()``

**Comment by Soullivaneuh** [comment]
@core23 In addition, it would be great to have a common abstract controller in core to avoid code duplication: https://github.com/sonata-project/SonataAdminBundle/blob/ff2375d3ab31f1d92e60dbe6427c8afff42fcd54/Controller/CRUDController.php#L1354

**Comment by core23** [comment]
Added a new Controller in the CoreBundle https://github.com/sonata-project/SonataCoreBundle/pull/199

File: Controller/PostController.php (hunk 2 above)
Comment (OskarStark) [inline Controller/PostController.php]: ping @core23

Comment (core23) [inline Controller/PostController.php]: Added request parameter and removed ``AbstractController``

**Comment by core23** [comment]
RTM @Soullivaneuh

**Comment by Soullivaneuh** [comment]
Thank you @core23

**Pull request merged**